test/unit/advertisement-operator/creation_test.go

```diff
@@ -0,0 +1,196 @@
+	saName := "fake-sa"
+	vkNamespace := "fake"
+	vkImage := "liqo/virtual-kubelet"
+	initVkImage := "liqo/init-vk"
+	homeClusterId := "cluster2"
+
+	deploy := advertisement_operator.CreateVkDeployment(adv, saName, vkNamespace, vkImage, initVkImage, homeClusterId)
+
+	assert.Equal(t, "vkubelet-"+adv.Spec.ClusterId, deploy.Name)
+	assert.Equal(t, vkNamespace, deploy.Namespace)
+	assert.Equal(t, advertisement_operator.GetOwnerReference(adv), deploy.OwnerReferences)
+	assert.Equal(t, adv.Spec.ClusterId, deploy.Spec.Template.Labels["cluster"])
+	assert.Len(t, deploy.Spec.Template.Spec.Volumes, 2)
+	assert.Equal(t, adv.Spec.KubeConfigRef.Name, deploy.Spec.Template.Spec.Volumes[0].VolumeSource.Secret.SecretName)
+	assert.Equal(t, initVkImage, deploy.Spec.Template.Spec.InitContainers[0].Image)
+	assert.Equal(t, vkImage, deploy.Spec.Template.Spec.Containers[0].Image)
+	assert.NotEmpty(t, deploy.Spec.Template.Spec.Containers[0].Args)
+	assert.Contains(t, deploy.Spec.Template.Spec.Containers[0].Args, adv.Spec.ClusterId)
+	assert.Contains(t, deploy.Spec.Template.Spec.Containers[0].Args, "vk-"+adv.Spec.ClusterId)
+	assert.Contains(t, deploy.Spec.Template.Spec.Containers[0].Args, vkNamespace)
+	assert.Contains(t, deploy.Spec.Template.Spec.Containers[0].Args, homeClusterId)
+	assert.NotEmpty(t, deploy.Spec.Template.Spec.Containers[0].Command)
+	assert.NotEmpty(t, deploy.Spec.Template.Spec.Containers[0].VolumeMounts)
+	assert.NotEmpty(t, deploy.Spec.Template.Spec.Containers[0].Env)
+	assert.Equal(t, saName, deploy.Spec.Template.Spec.ServiceAccountName)
+	assert.NotEmpty(t, deploy.Spec.Template.Spec.Affinity)
+}
+
+func TestCreateOrUpdate(t *testing.T) {
+	c := createFakeCRDClient()
+
+	testPod(t, c)
+	testAdvertisement(t, c)
+}
+
+func testPod(t *testing.T, c client.Client) {
+	name, ns := "pod", "fakens"
+	key := types.NamespacedName{
+		Name:      name,
+		Namespace: ns,
+	}
+	pod := createFakePod(name, ns)
+
+	// test pod creation
+	err := advertisement_operator.CreateOrUpdate(c, context.Background(), pod)
+	assert.Nil(t, err)
+
+	// creation requires some time to be effective
+	// wait 1 second to assure the resource is retrieved by Get
+	time.Sleep(1 * time.Second)
+
+	// test pod update
+	pod.Spec.Containers[0].Image = "image2"
+	err = advertisement_operator.CreateOrUpdate(c, context.Background(), pod)
+	assert.Nil(t, err)
+
+	time.Sleep(1 * time.Second)
+
+	var pod2 corev1.Pod
+	err = c.Get(context.Background(), key, &pod2)
+	assert.Nil(t, err)
+	assert.Equal(t, pod.Spec.Containers[0].Image, pod2.Spec.Containers[0].Image)
+}
+
+func testAdvertisement(t *testing.T, c client.Client) {
+	name, ns := "advertisement-cluster1", "fakens"
+	adv := createFakeAdv(name, ns)
+
+	// test adv creation
+	err := advertisement_operator.CreateOrUpdate(c, context.Background(), adv)
+	assert.Nil(t, err)
+
+	// creation requires some time to be effective
+	// wait 1 second to assure the resource is retrieved by Get
+	time.Sleep(1 * time.Second)
+
+	// test adv update
```

Comment:
Can you please wipe the following lines if they are commented, in order to keep the code clean.